Dip/Dip/Definition.swift

```diff
@@ -50,20 +50,62 @@ func ==(lhs: DefinitionKey, rhs: DefinitionKey) -> Bool {
 public enum ComponentScope {
   /// Indicates that a new instance of the component will be created each time it's resolved.
   case Prototype
+  /// Indicates that instances will be reused during resolve but will be discurded when topmost `resolve` method returns.
+  case ObjectGraph
   /// Indicates that resolved component should be retained by container and always reused.
   case Singleton
 }
 
 ///Definition of type T describes how instances of this type should be created when they are resolved by container.
 public final class DefinitionOf<T>: Definition {
```

Comment:
That will not work because if those methods on `DefinitionOf` will produce new copies then we will need to replace them in registry. We could solve that by having `register<T>(definition: DefinitionOf<T>)` and creating definition manually, but that will break initially nice syntax in separate steps without adding anything valuable.